ffs/filcold/filcold.go

```diff
@@ -3,6 +3,7 @@ package filcold
 import (
```

Comment:
TL;DR: I changed the _ColdStorage_ API `WaitForDeals` for `WaitForDeal`. If the caller want's to track multiple ones, it can do multiple calls concurrently. 

This greatly simplifies both this implementation, and gives more control to the caller.

Why was this necessary? Since now we support the client saying "cancel this executing job", the caller nice finer grained control about saving progress in individual deals is waiting for. So it can cancel the ones that are still waiting, and save the result of the ones that finished already.

In the previous implementations, since all deals were "waited in a single block", we couldn't solve that easily. That's saying, saving the ones that already finished and stopping the ongoing ones. They all were in the same `WaitForDeals` call, which was inconvenient.

The nice thing is that doing this change not only gives more control, but simplifies the code.